### PR TITLE
feat(dropdown): migrate dropdown component

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Migrating dropdown

**Solving issue**  
Azure ticket [AB#2385](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2385)

### CODE REVIEW

**Comment by EmelieLitwin** [comment]
Readme is still for dropdown-option

File: tegel/src/components/dropdown/dropdown.stories.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import readme from './readme.md';
```
Comment (mJarsater) [inline tegel/src/components/dropdown/dropdown.stories.js:1]: Maybe Stecil expects only one component per folder and therefor only creates on readme? In that case I guess we would have to create an internal folder structure to "hide" away the components subcomponents.

File: tegel/src/components/dropdown/dropdown.stories.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        options: { Default: 'default', Multiselect: 'multiselect' },
+      },
+      defaultValue: 'default',
+      description: 'Size of the dropdown',
+    },
+    size: {
+      name: 'Size',
+      control: {
+        type: 'radio',
+        options: { Large: 'lg', Medium: 'md', Small: 'sm' },
```
Comment (mJarsater) [inline tegel/src/components/dropdown/dropdown.stories.js:27]: This is nice! Means that we can get rid of a lot of switches in other components that assignes "Small" to "sm" for instance. 👍🏻

File: tegel/src/components/checkbox/checkbox.stories.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { formatHtmlPreview } from '../../utils/utils';
```
Comment (mJarsater) [inline tegel/src/components/checkbox/checkbox.stories.ts:1]: Should these changes be part of this PR?

File: tegel/src/components/dropdown/dropdown-filter.stories.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import { formatHtmlPreview } from '../../utils/utils';
```
Comment (mJarsater) [inline tegel/src/components/dropdown/dropdown-filter.stories.js:1]: Should this be a separate component in storybook? I would say that we include it under dropdown.

**Review by mJarsater — CHANGES_REQUESTED** [review]

File: tegel/src/components/checkbox/checkbox.stories.ts (hunk 3 above)
Comment (EmelieLitwin) [reply tegel/src/components/checkbox/checkbox.stories.ts:1]: Thank you, I missed this. Not sure what what happened there!

File: tegel/src/components/dropdown/dropdown-filter.stories.js (hunk 4 above)
Comment (EmelieLitwin) [reply tegel/src/components/dropdown/dropdown-filter.stories.js:1]: We talked a little about it at some point. The main reason is that it's its own component `<sdds-dropdown-filter>` unlike multiselect for example that is `<sdds-dropdown type='multiselect'>` plus it's got a separate readme.

File: tegel/src/components/dropdown/dropdown.stories.js (hunk 1 above)
Comment (EmelieLitwin) [reply tegel/src/components/dropdown/dropdown.stories.js:1]: I spent a some time finding different ways to trick stencil to generate a readme, but I was unsuccessful and felt like I had to move on but I definitely want to revisit it! At my last workplace all subcomponents had their own folders and were documented separately. Much easier 😆

**Comment by theJohnnyMe** [comment]
I am getting this on my local, please double-check: 

Screenshot 2022-10-31 at 11 48 11


Screenshot 2022-10-31 at 11 48 38

**Review by theJohnnyMe — APPROVED** [review]
There is an issue with the multi-select one, but let's keep it out of the scope of this task.

**Review by alexandereneroth — CHANGES_REQUESTED** [review]

File: tegel/src/components/dropdown/dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+    this.listItemArray = Array.from(this.host.children);
+    this.listItemArray.map(listItem => {
+      this.optionValues.push(listItem.value);
+      this.optionLabels.push(listItem.innerText);
+    });
+    this.setOptionFromOutside(this.defaultOption);
+
+    if (this.size === 'small') {
+      this.size = 'sm';
+      console.warn('size="small" is deprecated, use size="sm" instead');
```
Comment (alexandereneroth) [inline tegel/src/components/dropdown/dropdown.tsx:86]: nice!

File: tegel/src/components/dropdown/dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+  handleDocClick(ev) {
+    // To stop bubble click
+    ev.stopPropagation();
+    const target = ev ? ev.composedPath()[0] : window.event.target[0];
+    if (this.node !== undefined && this.node.contains(target)) {
+      if (typeof this.textInput !== 'undefined' || this.textInput === null) {
+        this.textInput.focus();
+      }
+      // TODO: Maybe can be this.open = !this.open ?
+      this.open ? !this.open : this.open;
```
Comment (alexandereneroth) [inline tegel/src/components/dropdown/dropdown.tsx:137]: This expression doesn't do anything, probably a bug here if we expect this to do something

**Review by alexandereneroth — CHANGES_REQUESTED** [review]

**Review by alexandereneroth — APPROVED** [review]
Nice work! ✨

**Pull request merged**